lib/radbeacon/le_scanner.rb

```diff
@@ -7,9 +7,26 @@ def initialize(duration = 5)
       @duration = duration
     end
 
+    def scan_command
+      rout, wout = IO.pipe
+      scan_command_str = "sudo hcitool lescan"
+      pid = Process.spawn(scan_command_str, :out => wout)
+      begin
+        Timeout.timeout(@duration) do
+          Process.wait(pid)
+        end
+      rescue Timeout::Error
+        Process.kill('INT', pid)
```

Comment:
:+1: This is a great use of `Timeout`.

But, I think you want `TERM` here.

- `SIGINT` is what is sent when a user presses Ctrl-C, the program interrupt signal
- `SIGTERM` is a termination signal, which is used between different processes to request process termination.